### PR TITLE
conditions component for creating filters

### PR DESCRIPTION
**Pull request by michaelzangerle**
- ui component to add, edit and remove conditions 

| Q             | A
| ------------- | ---
| Tests pass?   | yes
| Fixed tickets | none
| BC Breaks     | -
| Doc           | none

__Requires__
- https://github.com/sulu-io/sulu/pull/1058

### CODE REVIEW

File: src/Sulu/Bundle/ResourceBundle/Controller/FilterController.php [hunk 1]
```diff
@@ -69,15 +70,24 @@ function ($id) use ($locale) {
...
     public function cgetAction(Request $request)
     {
-        if ($request->get('flat') == 'true') {
-            $list = $this->getListRepresentation($request);
+        // check if context exists and filters are enabled for the given context
+        $context = $request->get('context');
+        $features = $this->getManager()->getFeaturesForAlias($context);
+
+        if ($context && (!$features || array_search('filters', $features) === false)) {
+            $exc = new RestException('Context unknown or filters not enabled for context!');
```
Comment (wachterjohannes) [inline src/Sulu/Bundle/ResourceBundle/Controller/FilterController.php]: you should add two diffrent exception classes for this (`UnkownContextException` and `NotEnabledFeatureException`)

File: src/Sulu/Bundle/ResourceBundle/Controller/FilterController.php [hunk 2]
```diff
@@ -69,15 +70,24 @@ function ($id) use ($locale) {
...
+            $view = $this->view($exc->toArray(), 400);
```
Comment (wachterjohannes) [inline src/Sulu/Bundle/ResourceBundle/Controller/FilterController.php]: if you return here then you can omit the else statement! it makes the code better readable and you don't need to stack if statements

File: src/Sulu/Bundle/ResourceBundle/DataFixtures/operators.xml [hunk 3]
```diff
@@ -88,18 +110,24 @@
             <translation locale="en">is not</translation>
         </translations>
     </operator>
-    <operator type="date" inputType="select" operator="between">
+    <operator type="date" inputType="select" operator="within">
```
Comment (wachterjohannes) [inline src/Sulu/Bundle/ResourceBundle/DataFixtures/operators.xml]: the other operators are uppercase. why is this lowercase?

File: src/Sulu/Bundle/ResourceBundle/Resource/Exception/FilterContextNotFoundException.php [hunk 4]
```diff
@@ -27,10 +27,10 @@ class FilterContextNotFoundException extends FilterException
      */
     private $name;
 
-    public function __construct($id)
```
Comment (wachterjohannes) [inline src/Sulu/Bundle/ResourceBundle/Resource/Exception/FilterContextNotFoundException.php]: is this exception for a missing "filter" feature in a context? maybe you could rename it to `MissingFeatureException` then it is reusable for upcoming features!

File: src/Sulu/Bundle/ResourceBundle/Resource/Exception/FilterContextNotFoundException.php [hunk 5]
```diff
@@ -27,10 +27,10 @@ class FilterContextNotFoundException extends FilterException
...
+    public function __construct($name)
     {
         $this->entityName = 'SuluResourceBundle:Filter';
-        $this->id = $id;
+        $this->id = $name;
```
Comment (wachterjohannes) [inline src/Sulu/Bundle/ResourceBundle/Resource/Exception/FilterContextNotFoundException.php]: you should rename property `id` to `name` if you rename the constructor argument!

File: src/Sulu/Bundle/ResourceBundle/Resources/config/services.xml [hunk 6]
```diff
@@ -53,6 +56,16 @@
         <service id="sulu_resource.operator_manager" class="%sulu_resource.operator_manager.class%">
             <argument type="service" id="sulu_resource.operator_repository"/>
         </service>
-    </services>
 
+
```
Comment (wachterjohannes) [inline src/Sulu/Bundle/ResourceBundle/Resources/config/services.xml]: i think here are two empty lines!

File: src/Sulu/Bundle/ResourceBundle/Resources/public/js/components/condition-selection/main.js [hunk 7]
```diff
@@ -0,0 +1,790 @@
...
+                case 'radio':
+                case 'checkbox':
+                    return createSelect.call(this, value, 'value', 'name', operator.values, constants.valueInputClass, true);
+                case '':
+                case 'simple':
+                    return createSimpleInput.call(this, value, constants.valueInputClass);
+                default:
+                    this.sandbox.logger.error('Input type "' + type + '" is not supported!');
+                    break;
+
```
Comment (wachterjohannes) [inline src/Sulu/Bundle/ResourceBundle/Resources/public/js/components/condition-selection/main.js]: empty line (=

File: src/Sulu/Bundle/ResourceBundle/DataFixtures/operators.xml (hunk 3 above)
Comment (michaelzangerle) [inline src/Sulu/Bundle/ResourceBundle/DataFixtures/operators.xml]: refactored the operators in the next PR (https://github.com/sulu-io/sulu/pull/1103) where I needed them to build the additional queries

Comment (wachterjohannes) [inline src/Sulu/Bundle/ResourceBundle/DataFixtures/operators.xml]: hmm thats not nice but i will take a look at it there!

**Pull request merged**